src/component.js

```diff
@@ -163,11 +163,27 @@ export default class Component {
         this.resolveBoundAttributes(el, false, extraVars)
     }
 
+    getEventNamesFromXAttrValue(value) {
+        const firstCharacter = value[0]
+        const lastCharacter = value[value.length]
```

Comment:
shouldn't this be 
```js
const lastCharacter = value[value.length - 1]
```
?

Reply:
Good catch. It definitely seems like I was sleeping there. I’ll give it another look

Reply:
Fixed with commit [464460d](https://github.com/alpinejs/alpine/commit/464460d).

Reply:
👍
Thanks for the great PR btw. Looking forward to seeing it land in master.

Reply:
Thanks! Happy to contribute